### ISSUE TITLE
Required components intializing after components that require it 

### ISSUE THREAD

**Issue by Testure** [Sleitnick/Knit#92]
Sometimes at seemingly random, a required component's Init function will be called **after** the component(s) that require it.

Somewhat helpful image of this happening: 

![Screenshot_770](https://user-images.githubusercontent.com/39919441/117094604-d2fd1d80-ad29-11eb-8573-87fc8e049aab.png)

**Comment by Sleitnick** [comment]
Hi @Testure, this is due to `Init` functions firing a frame _after_ the constructor. This does bring up a good question though: should the dependent components hold off on constructing until after the `Init` process is done?

**Comment by Testure** [comment]
In my use case, a the `GuiButtonComponent` creates signals that can be used by it's dependents, this happens in the `Init` function.
It would be typically expected that required components would be initialized before their dependents, so in my mind a solution would be to either implement lower priority for dependent components or to add a explanation of this behavior on the wiki. In the mean time I've worked around this by changing a variable once the `Init` function is done, and waiting for that.

**Comment by Sleitnick** [comment]
No longer relevant for this repository. If still an issue, please reopen in https://github.com/Sleitnick/RbxUtil for the Component module. However, the Component module is getting a full rewrite soon.
